pkg/publisher/publisher.go

```diff
@@ -11,11 +11,30 @@ import (
 
 var log = logrus.New()
 
-func Publish(ip net.IP, iface net.Interface, service Service, shutdown chan struct{}, waitGroup *sync.WaitGroup) (err error) {
+func Publish(iface net.Interface, service Service, shutdown chan struct{}, waitGroup *sync.WaitGroup) (err error) {
 	defer waitGroup.Done()
+	var v4addr, v6addr net.IP
+	addrs, err := iface.Addrs()
+	for _, addr := range addrs {
+		var currIP net.IP
+		switch v := addr.(type) {
+		case *net.IPNet:
+			currIP = v.IP
+		case *net.IPAddr:
+			currIP = v.IP
+		}
+		if currIP.To4() != nil {
```

Comment:
Does it not work to just do this version detection on the passed in IP and populate the appropriate field?

Reply:
I'm trying to have it expose both IPv4, and IPv6, and if the binding IP is specified as IPv6, that may not work. I guess we may just have to choose between exposing v4 and v6, but not both at the same time.